File: src/main/java/org/goafabric/calleeservice/logic/CalleeLogic.java
package org.goafabric.calleeservice.logic;

import org.goafabric.calleeservice.controller.Callee;
import org.springframework.stereotype.Component;

import java.util.concurrent.atomic.AtomicLong;

@Component
public class CalleeLogic {
    private final AtomicLong sleepTime = new AtomicLong(0L);

    public Callee sayMyName(String name) {
        sleep();
        return new Callee("0", "Your name is: " + name);
    }

    public Callee sayMyOtherName(String name) {
        sleep();
        return new Callee("0", "Your other name is: " + name);
    }

    public Callee save(Callee callee) {
        return new Callee("0", "Storing your message: " + callee.message());
    }

    public Callee setSleepTime(Long sleepTime) {
        this.sleepTime.set(sleepTime); //you should never change instance variables for a productive app
        return new Callee("0", "set sleepTime to:" + sleepTime);
    }

    private void sleep() {
        try {
            Thread.sleep(sleepTime.get());
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/java/org/goafabric/calleeservice/logic/CalleeLogic.kt
package org.goafabric.calleeservice.logic

import org.goafabric.calleeservice.service.Callee
import org.springframework.stereotype.Component
import java.util.concurrent.atomic.AtomicLong

@Component
class CalleeLogic {
    private val sleepTime = AtomicLong(0L)

    fun sayMyName (name : String) : Callee {
        sleep();
        return Callee(message = "Your name is: $name")
    }

    fun sayMyOtherName (name : String) : Callee {
        sleep();
        return Callee(message = "Your name is: $name")
    }

    fun setSleepTime(sleepTime : Long) : Callee {
        this.sleepTime.set(sleepTime)
        return Callee(message = "set sleepTime to: $sleepTime")
    }

    private fun sleep() {
        Thread.sleep(sleepTime.get())
    }
}